src/Microsoft.AspNetCore.SystemWebAdapters/Caching/Cache.cs

```diff
@@ -61,13 +62,21 @@ private static void AddChangeMonitors(CacheDependency? dependencies, CacheItemPo
 
     public void Insert(string key, object value) => _cache.Set(key, value, new CacheItemPolicy());
 
+    public void Insert(string key, object value, CacheDependency? dependencies)
+    {
+        var policy = new CacheItemPolicy();
+        AddChangeMonitors(dependencies, policy);
+        _cache.Set(key, value, policy);
+    }
+
     public void Insert(string key, object value, CacheDependency? dependencies, DateTime absoluteExpiration, TimeSpan slidingExpiration)
     {
         var policy = new CacheItemPolicy
         {
             AbsoluteExpiration = Convert(absoluteExpiration),
             SlidingExpiration = slidingExpiration,
         };
+        AddChangeMonitors(dependencies, policy);
 
         _cache.Set(key, value, policy);
```

Comment:
This may be good to have as a separate method as it's repeated throughout and would ensure that change monitors are added AND the cache is set.

Reply:
I've refactored this. If you don't like it as extension methods, I can move them internal to the class, but I think it looks cleaner as is.